C-Plus-Plus/cp/Intersection Point of Y shaped LinkedList.cpp

```diff
@@ -0,0 +1,233 @@
+    string list="";
+    temp1=head1; temp2=head2;
+    
+    while(temp1!=NULL){
+        length1++;
+        temp1=temp1->next1;
+    }
+    while(temp2!=NULL){
+       length2++;
+       temp2=temp2->next2;
+    }
+    if(length1>length2){
+        list="LIST 1";
+    }else{
+        list="LIST 2";
+    }
+    cout<<"***********************************************************************"<<endl;
+    cout<<"Since "<<list<<" is bigger in length."<<endl;
+    cout<<"Tell the position of node in "<<list<<" on which you want to intersect : "<<endl;
+    cout<<"***********************************************************************"<<endl;
+    
+    int pos;
+    cout<<"ENTER THE POSITION : ";
+    cin>>pos;
+    temp1=head1;temp2=head2;
+    
+    if(length1>length2){
+        while(temp2->next2!=NULL){
+            temp2=temp2->next2;
+        }
+        for(int i=0;i<pos-1;i++){
+            temp1=temp1->next1;
+        }
+        while(temp1!=NULL){
+            temp2->next2=temp1;
+            temp1=temp1->next1;
+            temp2=temp2->next2;
+        }
+        cout<<"Intersection Successfull"<<endl;
+    }
+    else{
+        while(temp1->next1!=NULL){
+            temp1=temp1->next1;
+        }
+        for(int i=0;i<pos-1;i++){
+            temp2=temp2->next2;
+        }
+        while(temp2!=NULL){
+            temp1->next1=temp2;
+            temp1=temp1->next1;
+            temp2=temp2->next2;
+        }
+        cout<<"Intersection Successfull"<<endl;
+    }
+}
+//--------------------------------------------------------------------------------------
+/*
+        Till now we have just work on intersecting two Linked List 
+        Now, the main aim of this program is in the next function to find the Intersection point.
+        
+        Approach to solve this problem :- 
+        ------------------------------
+        we will calculate the length of individual and will iterate the longer List till it's 
+        difference and then we wil iterate both the list together checking the pointer of next node 
+        is same or not.
+        
+*///---------------------------------------------------------------------------------------
+int Find_Intersection()
+{
+    int length1=0,length2=0;
+    temp1=head1; temp2=head2;
+    
+    while(temp1!=NULL){
+        length1++;
+        temp1=temp1->next1;             //finding length of list
+    }
+    while(temp2!=NULL){
+       length2++;
+       temp2=temp2->next2;
+    }
+    int  diff=abs(length1-length2);     //difference of both Lists
+    temp1=head1; temp2=head2;
+    
+    if(length1>length2){
+        for(int i=0;i<diff;i++){
+                temp1=temp1->next1;
+        }
+    }
+    else{
+         for(int i=0;i<diff;i++){
+                temp2=temp2->next2;
+        }
+    }
+    while(temp1!=NULL && temp2!=NULL){
+        if(temp1==temp2)
+        return temp1->data;
+        
+        temp1=temp1->next1;
+        temp2=temp2->next2;
+    }
+    return -1;
+}
```

Comment:
Add sample input/output in commented form, rest everything looks fine.

Reply:
Done adding, Please review again.